hpx/runtime/threads/detail/io_service_thread_pool.hpp

```diff
@@ -65,6 +65,13 @@ namespace hpx { namespace threads { namespace detail
         ///////////////////////////////////////////////////////////////////////
         void stop (std::unique_lock<compat::mutex>& l, bool blocking = true);
 
+        ///////////////////////////////////////////////////////////////////////
+        hpx::future<void> suspend(error_code& ec = throws);
```

Comment:
The API returning a future does not need to take an `error_code` as all possible exceptions will be reported through the future mechanism. Do I miss something?

Reply:
I don't think you're missing anything. This part of what I meant when I said it needs tidying up (just wanted to have the basics working first to try it out).

Reply:
Fair enough. Things start looking very good now. Thanks for your work on this!